src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Search.cs

```diff
@@ -0,0 +1,263 @@
+                {
+                    fValid = false;
+                    continue;
+                }
+            }
+
+            return fValid;
+        }
+        
+        internal static bool IsValidSearchableColumnType(DType type)
+        {
+            Contracts.AssertValid(type);
+
+            return type.Kind == DKind.String;
+        }
+
+        internal static bool IsColumnSearchable(DPath columnPath, IExternalDataSource dataSource)
+        {
+            Contracts.AssertValid(columnPath);
+            Contracts.AssertValue(dataSource);
+
+            var metadata = dataSource.DelegationMetadata?.FilterDelegationMetadata;
+            if (metadata == null)
+            {
+                return false;
+            }
+
+            return metadata.IsColumnSearchable(columnPath);
+        }
+
+        private bool IsValidSearchableColumnNode(TexlNode node, TexlBinding binding, IExternalDataSource dataSource)
+        {
+            Contracts.AssertValue(node);
+            Contracts.AssertValue(binding);
+            Contracts.AssertValue(dataSource);
+
+            if (binding.ErrorContainer.HasErrors(node) || node.Kind != NodeKind.StrLit)
+            {
+                return false;
+            }
+
+            StrLitNode columnNode = Contracts.VerifyValue(node.AsStrLit());
+            string columnName = columnNode.Value;
+            if (string.IsNullOrEmpty(columnName) || !IsColumnSearchable(DPath.Root.Append(new DName(columnName)), dataSource))
+            {
+                SuggestDelegationHint(node, binding);
+                return false;
+            }
+
+            return true;
+        }
+
+        public override bool UpdateDataQuerySelects(CallNode callNode, TexlBinding binding, DataSourceToQueryOptionsMap dataSourceToQueryOptionsMap)
+        {
+            Contracts.AssertValue(callNode);
+            Contracts.AssertValue(binding);
+
+            // Ignore delegation warning
+            if (!CheckArgsCount(callNode, binding, DocumentErrorSeverity.Moderate))
+            {
+                return false;
+            }
+
+            var args = Contracts.VerifyValue(callNode.Args.Children);
+            DType dsType = binding.GetType(args[0]);
+            if (dsType.AssociatedDataSources == null)
+            {
+                return false;
+            }
+
+            int cargs = args.Count;
+            bool retval = false;
+
+            for (int i = 2; i < cargs; i++)
+            {
+                string columnName;
+                DType columnType;
+                if (binding.Features.SupportColumnNamesAsIdentifiers)
+                {
+                    columnName = ((FirstNameNode)args[i]).Ident.Name;
+                    columnType = dsType.GetType(new DName(columnName));
+                }
+                else
+                {
+                    columnType = binding.GetType(args[i]);
```

Comment:
>columnType [](http://example.com/codeflow?start=20&length=10)

We already validated that the column types should be string (in CheckTypes), so there's no need to ask the binder again.